server/services/split_service.py

```diff
@@ -0,0 +1,141 @@
+                    new_y = y * 2 + j
+                    new_zoom = zoom + 1
+                    new_square = SplitService._create_square(new_x, new_y, new_zoom)
+                    feature = geojson.Feature()
+                    feature.geometry = new_square
+                    feature.properties = {
+                        'x': new_x,
+                        'y': new_y,
+                        'zoom': new_zoom,
+                        'splittable': True
+                    }
+                    split_geoms.append(feature)
+
+            return split_geoms
+        except Exception as e:
+            raise SplitServiceError(f'unhandled error splitting tile: {str(e)}')
+
+    @staticmethod
+    def _create_square(x, y, zoom) -> geojson.MultiPolygon:
+        """
+        Function for creating a geojson.MultiPolygon square representing a single OSM tile grid square
+        :param x: osm tile grid x
+        :param y: osm tile grid x
+        :param zoom: osm tile grid zoom level
+        :return: eojson.MultiPolygon in EPSG:4326
+        """
+        # Maximum resolution
+        MAXRESOLUTION = 156543.0339
+
+        # X/Y axis limit
+        max = MAXRESOLUTION * 256 / 2
+
+        # calculate extents
+        step = max / (2 ** (zoom - 1))
+        xmin = x * step - max
+        ymin = y * step - max
+        xmax = (x + 1) * step - max
+        ymax = (y + 1) * step - max
+
+        # make a shapely multipolygon
+        multipolygon = MultiPolygon([Polygon([(xmin, ymin), (xmax, ymin),
+                                              (xmax, ymax), (xmin, ymax)])])
+
+        # use the database to transform the geometry from 3857 to 4326
+        transformed_geometry = ST_Transform(shape.from_shape(multipolygon, 3857), 4326)
+
+        # use DB to get the geometry as geojson
+        return geojson.loads(db.engine.execute(transformed_geometry.ST_AsGeoJSON()).scalar())
+
+    @staticmethod
+    def split_task(split_task_dto: SplitTaskDTO) -> TaskDTOs:
+        """
+        Replaces a task square with 4 smaller tasks at the next OSM tile grid zoom level
+        Validates that task is:
+         - locked for mapping by current user
+         - splittable (splittable property is True)
+        :param split_task_dto:
+        :return: new tasks in a DTO
+        """
+        # get the task to be split
+        original_task = Task.get(split_task_dto.task_id, split_task_dto.project_id)
+        if original_task is None:
+            raise NotFound()
+
+        # check its locked for mapping by the current user
+        if TaskStatus(original_task.task_status) != TaskStatus.LOCKED_FOR_MAPPING:
+            raise SplitServiceError('Status must be LOCKED_FOR_MAPPING to split')
+
+        if original_task.locked_by != split_task_dto.user_id:
+            raise SplitServiceError('Attempting to split a task owned by another user')
+
+        # check it's splittable
+        if not original_task.splittable:
```

Comment:
I'd move this to line 99, if task is not splittable there's no need to do any further checking.